src/components/organisms/BannerAdmin/BannerAdmin.tsx

```diff
@@ -1,90 +1,176 @@
-import React, { useState, useCallback, useRef } from "react";
+import React, { useCallback, useState, useEffect } from "react";
+import { useForm } from "react-hook-form";
 
 import { makeUpdateBanner } from "api/bannerAdmin";
 
 import { AnyVenue } from "types/venues";
+import { Banner } from "types/banner";
+
+import { useShowHide } from "hooks/useShowHide";
+
+import { Button } from "components/atoms/Button";
+import { Checkbox } from "components/atoms/Checkbox";
+import { InputField } from "components/atoms/InputField";
+
+import { ConfirmationBannerModal } from "./ConfirmationBannerModal";
+
+import "./BannerAdmin.scss";
 
 interface BannerAdminProps {
   venueId?: string;
   venue: AnyVenue;
+  onClose: () => void;
 }
 
 // @debt This component is almost exactly the same as IframeAdmin, we should refactor them both to use the same generic base component
 //   BannerAdmin is the 'canonical example' to follow when we do this
-export const BannerAdmin: React.FC<BannerAdminProps> = ({ venueId, venue }) => {
-  const existingBannerMessage = venue?.bannerMessage ?? "";
-
-  const inputFieldRef = useRef<HTMLInputElement>(null);
-  const [error, setError] = useState<string | null>();
+export const BannerAdmin: React.FC<BannerAdminProps> = ({
+  venueId,
+  venue,
+  onClose,
+}) => {
+  const { register, handleSubmit, errors, reset, watch } = useForm<Banner>({
+    mode: "onChange",
+    reValidateMode: "onChange",
+  });
+  const isActionButtonValue = watch("isActionButton");
+
+  const [bannerData, setBannerDate] = useState<Banner>();
+
+  const {
+    isShown: isShowModal,
+    show: showModal,
+    hide: hideModal,
+  } = useShowHide();
```

Comment:
What modal does this refer to? Should be `showFooModal`, `hideFooModal`, `isFooModalVisible`

Reply:
renamed 771b70a